app/src/main/java/com/portfolio/tasky/entry/repositories/EntryRepositoryImpl.kt

```diff
@@ -0,0 +1,59 @@
+package com.portfolio.tasky.entry.repositories
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.portfolio.tasky.entry.network.EntryApiCall
+import com.portfolio.tasky.entry.models.AuthenticationRequest
+import com.portfolio.tasky.entry.models.AuthenticationResponse
+import com.portfolio.tasky.entry.models.RegisterRequest
+import com.portfolio.tasky.networking.usecases.domain.TaskyCallStatus
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import retrofit2.Response
+import javax.inject.Inject
+
+
+class EntryRepositoryImpl@Inject constructor(private val entryApiCall: EntryApiCall, private val taskyCallStatus: TaskyCallStatus) : EntryRepository  {
+    private val registrationLiveData = MutableLiveData<Boolean>()
+    private val loginLiveData = MutableLiveData<AuthenticationResponse?>()
+
+    private val coroutineExceptionHandler = CoroutineExceptionHandler{ _, throwable ->
+        throwable.printStackTrace()
+    }
+
+    override fun doRegistration(registerRequest: RegisterRequest): MutableLiveData<Boolean> {
+
+        CoroutineScope(Dispatchers.IO + coroutineExceptionHandler).launch {
```

Comment:
Reason to call coroutine here is because I wanted to observe proper instance of LiveDatas in Login and Registration fragment.